### ISSUE TITLE
Continous Reporting for Colors (extension of `OSC 10`, …)

### ISSUE THREAD

**Issue by bash** [alacritty/alacritty#7900]
Hi 👋🏼 

Terminal programs can currently query the terminal's colors (foreground, background, etc.) by sending the color's corresponding `OSC` sequence with `?` instead of a color. \
*Example*: `OSC 10; ? ST` queries the terminal for its foreground color.

My goal is to extend this querying mechanism with a way for long lived applications to be notified when the color changes. 

This enables editors (e.g. vim) and multiplexers (e.g. tmux, zellij) to react to color changes. This is particularly interesting when the user has set up Alacritty to react to the OS's dark-light preference.

Based on [lots of input](https://gitlab.gnome.org/GNOME/vte/-/issues/2740) from Egmont Koblinger (VTE) I put together [a document](https://github.com/bash/continuous-color-reporting) that describes the mechanism in detail.

**tl;dr** Continuous reporting can be enabled individually for each color using the same `OSC` sequence as for one-time querying but passing `+?` instead of `?`. Reports work very similar to FocusIn/FocusOut events.
*Example*: `OSC 10; +? ST` enables continous reporting for the foreground color.

I would love to hear what your thoughts on my specification are and if this is something that you would be open to include in Alacritty. I would be more than happy to work on a PR for this.

Some very broad signals (e.g. we would like a mechanism for this, but not this specific solution) would already help a lot :)

**Comment by chrisduerr** [comment]
> I would love to hear what your thoughts on my specification are and if this is something that you would be open to include in Alacritty. I would be more than happy to work on a PR for this.

I'm not interested in this at all. I see no reason why the foreground process would need to be notified about color updates when it should be the only one actually requesting color updates. That seems like a serious issue in the way the client is written, not a reason to add new escape mechanisms.

Having this included in the `OSC` also doesn't make any sense. It sounds more like a new terminal mode to me that would report all color changes. Having a mode change like that encoded in an existing OSC escape just feels lazy.

**Comment by bash** [comment]
> I'm not interested in this at all. 

Is it the specific mechanism or the general goal of notifying processes about color changes?

> I see no reason why the foreground process would need to be notified about color updates when it should be the only one actually requesting color updates.

The foreground process doesn't know *when* the color changes.

Currently the only option is to poll for color updates by repeatedly sending something like `OSC 10; ? ST`. That seems like a very wasteful solution to me.

Would you be more open to a less granular mechanism (for example a mode that is enabled/disabled with `DECSET`/`DECRST`)?

**Comment by chrisduerr** [comment]
> Is it the specific mechanism or the general goal of notifying processes about color changes?

Both.
